Problems with local build instructions on Mac (and perhaps other 'nix platforms)

It turns out that the instructions for doing local builds ( at least on a Mac) don't work. The steps I've gone through so far to make progress (with links to some of the relevant roadblocks) are:

+ Install command line tools (`xcode-select --install`) (may not be necessary depending on how recent your last OS update was).
+ Update `gem` itself (see [this in-progress update to the jupyter-book instructions](https://github.com/mwcraig/jupyter-book/commit/a9f9d62af61f6b819eee82b3ca83685e5acb8d95), which includes a link to a [statement from the bundler folks](https://bundler.io/blog/2019/01/04/an-update-on-the-bundler-2-release.html) and an [answer on stackoverflow](https://stackoverflow.com/a/54118329)). **THIS IS NOT MAC-SPECIFIC**
+ Install the conda compilers for mac: `conda install clangxx_osx-64` (see https://stackoverflow.com/questions/47314899/how-to-avoid-clang-no-such-file-or-directory-error). **My hunch is that something like this is needed on linux too.**
+ On 10.14/Mojave, make the standard headers show up in the standard places because [apple just had to move them](https://github.com/neovim/neovim/issues/9050#issuecomment-424417456).

Once that stuff is done, `make install` proceeds much farther, until it gets to building `eventmachine`, which uses openssl, and it can't find the headers for that:

```
./project.h:119:10: fatal error: 'openssl/ssl.h' file not found
#include <openssl/ssl.h>
         ^~~~~~~~~~~~~~~
```

That led me to [this anaconda issue](https://github.com/ContinuumIO/anaconda-issues/issues/9863) about building gems. I tried the monkey patch at https://github.com/ContinuumIO/anaconda-issues/issues/9863#issuecomment-460512146 but it didn't do the trick for me.

Will continue trying for a little bit longer and report back if I get something to work.

--- a/jupyter_book/book_template/content/guide/03_build.md
+++ b/jupyter_book/book_template/content/guide/03_build.md
@@ -69,6 +69,26 @@ make install
 which will install Bundler (a Ruby depency management tool) and then install the plugins
 needed to build the site for your book.
 
+#### If `make install` fails
+
+If running `make install` fails with this error
+
+```console
+`find_spec_for_exe': can't find gem bundler (>= 0.a) with executable bundle (Gem::GemNotFoundException)
+```
+
+it is because of a [version mismatch between `gem` and `bundler`](https://bundler.io/blog/2019/01/04/an-update-on-the-bundler-2-release.html). You can fix the problem by [updating gem](https://stackoverflow.com/a/54118329)
+
+```bash
+gem update --system
+```
+
+and then re-running
+
+```bash
+make install
+```
+
 ### Build the site HTML for your book
 
 Once you've generated the markdown for your notebooks and installed the

--- a/jupyter_book/book_template/content/guide/03_build.md
+++ b/jupyter_book/book_template/content/guide/03_build.md
@@ -60,34 +60,42 @@ the *`conda`* package manager:
 ```bash
 conda install -c conda-forge ruby
 ```
-Once you have Ruby installed, run
+
+Once you have Ruby installed, the conda-built clang compiler for your system need to be installed. Details depend on your OS.
+
+#### On OSX
+
+Two steps are needed on OSX. First, install the *`conda`'s* clang compiler:
+
+```bash
+conda install -c conda-forge clangxx_osx-64
+```
+
+If you are running OSX 10.14 (Mojave) you also need to install system libraries in
+the default *nix locations by running
+
+```bash
+open /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg
+```
+
+#### On Linux
+
+Install *`conda`'s* clang compiler:
+
+```bash
+gxx_linux-64
+```
+
+#### Install Ruby plugins
+
+Finally, run
 
 ```bash
 make install
 ```
 
-which will install Bundler (a Ruby depency management tool) and then install the plugins
+which will install Bundler (a Ruby dependency management tool) and then install the plugins
 needed to build the site for your book.
-
-#### If `make install` fails
-
-If running `make install` fails with this error
-
-```console
-`find_spec_for_exe': can't find gem bundler (>= 0.a) with executable bundle (Gem::GemNotFoundException)
-```
-
-it is because of a [version mismatch between `gem` and `bundler`](https://bundler.io/blog/2019/01/04/an-update-on-the-bundler-2-release.html). You can fix the problem by [updating gem](https://stackoverflow.com/a/54118329)
-
-```bash
-gem update --system
-```
-
-and then re-running
-
-```bash
-make install
-```
 
 ### Build the site HTML for your book
 

